src/dill/anySwap/AnyCallClient.sol

```diff
@@ -0,0 +1,532 @@
+    bytes32 public constant PAUSE_SWAPIN_ROLE = keccak256("PAUSE_SWAPIN_ROLE");
+    bytes32 public constant PAUSE_FALLBACK_ROLE =
+        keccak256("PAUSE_FALLBACK_ROLE");
+
+    uint256 public ANYCALL_FLAGS = 2; // Fee on SRC (mainnet) chain
+    string public APP_ID = "0"; // Could be changed
+
+    // associated tokens on each chain
+    mapping(address => mapping(uint256 => address)) public tokenPeers;
+
+    event LogSwapout(
+        address indexed token,
+        address indexed sender,
+        address indexed receiver,
+        uint256 amount,
+        uint256 toChainId,
+        int256[] weights
+    );
+    event LogSwapin(
+        address indexed token,
+        address indexed sender,
+        address indexed receiver,
+        uint256 amount,
+        uint256 fromChainId,
+        int256[] weights
+    );
+    event LogSwapoutFail(
+        address indexed token,
+        address indexed sender,
+        address indexed receiver,
+        uint256 amount,
+        uint256 toChainId,
+        int256[] weights
+    );
+
+    modifier onlyDistributor() {
+        require(msg.sender == distributor, "AnycallClient: onlyDistributor");
+        _;
+    }
+
+    constructor(
+        address _admin,
+        address _callProxy,
+        address _distributor
+    ) AnycallClientBase(_admin, _callProxy) {
+        distributor = _distributor;
+    }
+
+    function setTokenPeers(
+        address srcToken,
+        uint256[] calldata chainIds,
+        address[] calldata dstTokens
+    ) external onlyAdmin {
+        require(chainIds.length == dstTokens.length);
+        for (uint256 i = 0; i < chainIds.length; i++) {
+            tokenPeers[srcToken][chainIds[i]] = dstTokens[i];
+        }
+    }
+
+    function setDistributor(address _distributor) external onlyAdmin {
+        require(_distributor != address(0));
+        distributor = _distributor;
+    }
+
+    /// @dev Call by the user to submit a request for a cross chain interaction
+    function bridge(
+        address token,
+        uint256 amount,
+        address receiver,
+        uint256 toChainId,
+        int256[] memory weights,
+        uint256 periodId
+    ) external payable //onlyDistributor whenNotPaused(PAUSE_SWAPOUT_ROLE) 
+    {
+        address clientPeer = clientPeers[toChainId];
+        require(clientPeer != address(0), "AnycallClient: no dest client");
+
+        address dstToken = tokenPeers[token][toChainId];
+        require(dstToken != address(0), "AnycallClient: no dest token");
+
+        uint256 oldCoinBalance;
+        if (msg.value > 0) {
+            oldCoinBalance = address(this).balance - msg.value;
+        }
+
+        address _underlying = _getUnderlying(token);
+        if (
+            _underlying != address(0) &&
+            IERC20(token).balanceOf(msg.sender) < amount
+        ) {
+            uint256 old_balance = IERC20(_underlying).balanceOf(token);
+            IERC20(_underlying).safeTransferFrom(msg.sender, token, amount); 
```

Comment:
I don't believe that the user has the `amount` to transfer in this case. It should be `IERC20(token).balanceOf(msg.sender)` instead